classes/Log/Logger.php

```diff
@@ -175,9 +175,37 @@ public function updateLogsPath(string $logsPath): void
      */
     public function log($level, string $message, array $context = []): void
     {
+        $className = $this->getCallingClass();
+
         if (is_resource($this->fd)) {
-            fwrite($this->fd, '[' . date('Y-m-d H:i:s') . '] ' . self::$levels[$level] . ' - ' . $message . PHP_EOL);
+            fwrite(
+                $this->fd,
+                '[' . date('Y-m-d H:i:s') . '] ' . self::$levels[$level] . ' - ' .
+                ($className ? $className . ' - ' : '') .
+                $message . PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Get the name of the class that called the logger.
+     */
+    private function getCallingClass(): ?string
+    {
+        $trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS);
```

Comment:
I loved the idea but I didn't know this method would be used, it is known to take a long time to return the stack. Can you please check the impact on the performance?

Another solution would be to set the class of the running task with a getter / setter.